tests/test_one_of.py

```diff
@@ -227,5 +231,37 @@ def test_one_of(self):
         self.assertAlmostEqual(counts[2] / 10000, 0.25, delta=1.0)
 
 
+TEST_ONEOF_EXTENDED_TEST_CASES = [
+    [None, tuple()],
+    [None, (sa.Rotate(np.pi / 8),)],
+    [None, (sa.Flip(0), sa.Flip(1), sa.Rotate90(1), sa.Zoom(0.8), ia.NormalizeIntensity())],
+    [("a",), (sd.Rotated(("a",), np.pi / 8),)],
+]
+
+
+class TestOneOfAPITests(unittest.TestCase):
+    @staticmethod
+    def data_from_keys(keys):
+        if keys is None:
+            data = torch.unsqueeze(torch.tensor(np.arange(12 * 16).reshape(12, 16)), dim=0)
+        else:
+            data = {}
+            for i_k, k in enumerate(keys):
+                data[k] = torch.unsqueeze(torch.tensor(np.arange(12 * 16)).reshape(12, 16) + i_k * 192, dim=0)
+        return data
+
+    @parameterized.expand(TEST_ONEOF_EXTENDED_TEST_CASES)
+    def test_execute_change_start_end(self, keys, pipeline):
+        data = self.data_from_keys(keys)
+
+        with self.assertRaises(ValueError):
+            c = OneOf(deepcopy(pipeline))
+            c(data, start=1)
+
+        with self.assertRaises(ValueError):
+            c = OneOf(deepcopy(pipeline))
+            c(data, end=1)
```

Comment:
```suggestion
        c = OneOf(deepcopy(pipeline))
        with self.assertRaises(ValueError):
            c(data, start=1)

        with self.assertRaises(ValueError):
            c = OneOf(deepcopy(pipeline))
            c(data, end=1)
```

Reply:
Done slightly differently to the suggestion but following the advice